axon-server-connector/src/main/java/org/axonframework/axonserver/connector/query/AxonServerQueryBus.java

```diff
@@ -341,7 +341,8 @@ public void disconnect() {
      */
     @ShutdownHandler(phase = Phase.OUTBOUND_QUERY_CONNECTORS)
     public CompletableFuture<Void> shutdownDispatching() {
-        return shutdownLatch.initiateShutdown();
+        return shutdownLatch.initiateShutdown()
+                .thenRun(()-> updateEmitter.complete(p-> true));
```

Comment:
This doesn't seem corrdect. Shutting down an application doesn't imply that all subscription queries are completed. Other nodes would still be able to provide updates to the stream

Reply:
Thanks for pointing it out. I have reworked the code to just close active registrations instead of completing the whole subscription query.